lib/adapters/http.js

```diff
@@ -181,17 +181,21 @@ module.exports = function httpAdapter(config) {
 
       // uncompress the response body transparently if required
       var stream = res;
-      switch (res.headers['content-encoding']) {
-      /*eslint default-case:0*/
-      case 'gzip':
-      case 'compress':
-      case 'deflate':
+
+      // if no content or decompress disabled we should not decompress
+      if (res.statusCode !== 204 && config.decompress !== false) {
```

Comment:
Can you update the condition with #2666?

Reply:
I added the HEAD condition from #2666